src/controllers/user.register.ts

```diff
@@ -0,0 +1,58 @@
+import 'dotenv/config';
+import { Request, Response } from 'express';
+import User from '../models/user.model';
+import { signJwt } from '../utils/authentication';
+import { hashPassword } from '../utils/hashPassword';
+import { validateData } from '../utils/validateData';
+import { errorHandler } from '../utils/errorHandler';
+import LoginRegisterRepository from '../repositories/loginRegister.repository';
+
+export const registerUser = async (req: Request, res: Response): Promise<Response | undefined> => {
+
+    const {
+        username,
+        email,
+        password,
+        gender,
+        region
+    }:User = req.body;
+
+    try {
+
+        const isValid = validateData(email, password);
+        if(!isValid.valid){
+            return res.status(400).send(isValid);
+        };
+        
+        const register = new LoginRegisterRepository();
```

Comment:
Controller calling Repository ??? Controller -> Service -> Repository

Reply:
I only use class LoginRegisterRepository in controller to call their methods, and they are for database interaction. Do I need to make service folder for services, like one file for login and register and than call repo functions there?

Reply:
Yup service will proxy repo methods if there is no extra logic. Its design mate